### PR TITLE
Introducing okta cli

### PR DESCRIPTION
**Pull request by dogeared**
This blog post has the following features:

- [x] A title that's approved by Dev Marketing
- [x] A URL that's approved by Dev Marketing
- [X] Rendered locally and confirmed that no Markdown typos exist

### CODE REVIEW

File: _source/_posts/2020-10-04-introducing-okta-cli.md [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+
+Now, it looks like this:
+
+```bash
+okta start
+```
+
+I want to make this a little more challenging for myself, so I am going to use a .NET sample application, since I am a Java developer. Here's the output of the above command:
+
+```bash
```
Comment (bdemers) [inline _source/_posts/2020-10-04-introducing-okta-cli.md:74]:
```suggestion
```txt
```

File: _source/_posts/2020-10-04-introducing-okta-cli.md [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+Don't EVER commit okta-aspnetcore-mvc-example/appsettings.json into source control
+Run this application with:
+    dotnet run --project okta-aspnetcore-mvc-example
+```
+
+There's a lot going on here from the humble `okta start` command, so let's break it down.
+
+First I see a one-time requirement in order to provision the Okta org.
+
+I put in some minimal information: name, email and company. In about 20 seconds, a shiny new Okta org is allocated for me and the Okta CLI shows me my Org URL (https://dev-159750.okta.com).
```
Comment (bdemers) [inline _source/_posts/2020-10-04-introducing-okta-cli.md:114]:
```suggestion
I put in some minimal information: name, email and company. In about 20 seconds, a shiny new Okta org is allocated for me and the Okta CLI shows me my Org URL (`https://dev-159750.okta.com`).
```

File: _source/_posts/2020-10-04-introducing-okta-cli.md [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+
+Next, I need to prove that I own the email address I entered. an One-Time Passcode (OTP) is sent to my email address and the Okta CLI waits for me to paste that in.
+
+{% img blog/introducing-okta-cli/okta-email.png alt:"okta email" width:"600" %}{: .center-image }
+
+Once my email address is verified, the Okta CLI displays a unique link to set my password.
+
+> **NOTE:** The order-of-operations here is not important. You can set your password before you choose a sample or afterwards. But, you should set your password in a timely fashion as that unique URL expires after 24 hours.
```
Comment (bdemers) [inline _source/_posts/2020-10-04-introducing-okta-cli.md:122]: Good call out!

File: _source/_posts/2020-10-04-introducing-okta-cli.md [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+
+Next, I see a list of available samples. As of this writing, there are four samples, but we are expanding that list. And, the Okta CLI automatically makes any new samples available in the list. You can create as many sample as you like with your Okta Org. I chose option 3: `ASP.net Core 4 MVC`
```
Comment (bdemers) [inline _source/_posts/2020-10-04-introducing-okta-cli.md:124]: As man examples as you like, as long at it's <=5...

File: _source/_posts/2020-10-04-introducing-okta-cli.md [hunk 5]
```diff
@@ -0,0 +1,319 @@
...
+
+Finally, I see information about the Application that was provisioned in my Okta Org along with instructions to run the sample app.
```
Comment (bdemers) [inline _source/_posts/2020-10-04-introducing-okta-cli.md:126]:
```suggestion
Finally, I see information about the OIDC Application that was provisioned in my Okta Org along with instructions to run the sample app.
```

File: _source/_posts/2020-10-04-introducing-okta-cli.md [hunk 6]
```diff
@@ -0,0 +1,319 @@
...
+
+info: Microsoft.Hosting.Lifetime[0]
+      Now listening on: https://localhost:5001
+info: Microsoft.Hosting.Lifetime[0]
+      Application started. Press Ctrl+C to shut down.
+...
+```
+
+> Did you know you could run dotnet on macOS? What a world! Simply run: `brew cask install dotnet-sdk`
+
```
Comment (bdemers) [inline _source/_posts/2020-10-04-introducing-okta-cli.md:154]: Maybe mention a private/incognito window, to make it more obvious the user has logged in.

File: _source/_posts/2020-10-04-introducing-okta-cli.md [hunk 7]
```diff
@@ -0,0 +1,319 @@
...
+Click **Sign In** and you're redirected over to Okta to authenticate. After authenticating, you're redirected back to the app and it knows who you are!
+
+{% img blog/introducing-okta-cli/dotnet-authenticated.png alt:"dotnet authenticated" width:"600" %}{: .center-image }
+
+> **NOTE:** If you didn't set your password earlier, you'll need to now. Follow the unique link shown in the Okta CLI output. In my case, it was: `https://dev-159750.okta.com/reset_password/drpdRFTmBi8yQ-uIzbpx?fromURI=/`
+
+## Looking Under the Hood of the Okta CLI
+
+I don't know much at all about .NET programming. I couldn't tell you the difference between .NET Core and ASP.NET or the differences between 3 and 4 (I think there's a 5 coming?) But, in less than 5 minutes, I had a running app that I could authenticate to using Okta.
```
Comment (bdemers) [inline _source/_posts/2020-10-04-introducing-okta-cli.md:167]: This is a great statement!

File: _source/_posts/2020-10-04-introducing-okta-cli.md [hunk 8]
```diff
@@ -0,0 +1,319 @@
...
+
+At this point, you could fire up your app as normal and have it fully integrated with Okta for authentication:
+
+```bash
+./mvnw spring-boot:run
+```
+
+The important bits to set up a project for use with the Okta CLI are:
+
+1. Add the `OKTA_CLI_` placeholder values to the configuration files that make sense for your language and framework
```
Comment (bdemers) [inline _source/_posts/2020-10-04-introducing-okta-cli.md:284]:
```suggestion
1. Add the `OKTA_CLI_*` placeholder values to the configuration files that make sense for your language and framework
```

**Review by bdemers — APPROVED** [review]
A few minor nits

File: _source/_posts/2020-10-04-introducing-okta-cli.md [hunk 9]
```diff
@@ -0,0 +1,319 @@
...
+    "javascript",
+    "mobile",
+    "php",
+    "python",
+    "ruby",
+    "security",
+    "devops"
+]
+description: "Use the Okt CLI to get started in under 5 minutes!"
+tags: [security, oauth, oauth2, oidc, openid-connect]
```
Comment (bdemers) [inline _source/_posts/2020-10-04-introducing-okta-cli.md:19]:
```suggestion
tags: [security, oauth, oauth2, oidc, openid-connect, cli]
```

**Comment by mraible** [comment]
@dogeared Do you still plan on publishing this at some point? If so, we should get it scheduled!

File: _source/_posts/2020-10-04-introducing-okta-cli.md [hunk 10]
```diff
@@ -0,0 +1,319 @@
...
+./mvnw spring-boot:run
+```
+
+The important bits to set up a project for use with the Okta CLI are:
+
+1. Add the `OKTA_CLI_` placeholder values to the configuration files that make sense for your language and framework
+2. Add the `.okta/sample-config.yaml` file with the right application type for your language and framework
+3. Run `okta start` to provision and Okta app and update all the placeholders in your config file(s).
+
+> The easiest way to see what configuration you need for your language and framework is to visit the official samples found in the [okta-samples](https://github.com/okta-samples) GitHub org.
```
Comment (mraible) [inline _source/_posts/2020-10-04-introducing-okta-cli.md:288]: "official" samples or "community" samples? 🙂

File: _source/_posts/2020-10-04-introducing-okta-cli.md [hunk 11]
```diff
@@ -0,0 +1,319 @@
...
+
+Currently, the available placeholder values are:
+
+| Placeholder            | Description                                                                                              |
+|------------------------|----------------------------------------------------------------------------------------------------------|
+| CLI_OKTA_ORG_URL       | The base URL to your Okta Org                                                                            |
+| CLI_OKTA_ISSUER        | The URL to your Okta OAuth 2.0 [Authorization Server](https://developer.okta.com/docs/concepts/auth-servers/) |
+| CLI_OKTA_CLIENT_ID     | The client id set by Okta for [web](https://developer.okta.com/docs/guides/implement-auth-code/overview/), [native](https://developer.okta.com/docs/guides/implement-auth-code-pkce/overview/), [browser](https://developer.okta.com/docs/guides/implement-auth-code-pkce/overview/) and [service](https://developer.okta.com/docs/guides/implement-client-creds/overview/) type apps                                 |
```
Comment (mraible) [inline _source/_posts/2020-10-04-introducing-okta-cli.md:296]:
```suggestion
| CLI_OKTA_CLIENT_ID     | The client ID set by Okta for [web](https://developer.okta.com/docs/guides/implement-auth-code/overview/), [native](https://developer.okta.com/docs/guides/implement-auth-code-pkce/overview/), [browser](https://developer.okta.com/docs/guides/implement-auth-code-pkce/overview/) and [service](https://developer.okta.com/docs/guides/implement-client-creds/overview/) type apps                                 |
```

File: _source/_posts/2020-10-04-introducing-okta-cli.md [hunk 12]
```diff
@@ -0,0 +1,319 @@
...
+| CLI_OKTA_CLIENT_SECRET | The client secret set by Okta for [web](https://developer.okta.com/docs/guides/implement-auth-code/overview/) and [service](https://developer.okta.com/docs/guides/implement-client-creds/overview/) type apps                                              |
+
```
Comment (mraible) [inline _source/_posts/2020-10-04-introducing-okta-cli.md:298]: Is this table viewable on mobile?

File: _source/_posts/2020-10-04-introducing-okta-cli.md [hunk 13]
```diff
@@ -0,0 +1,319 @@
...
+## Adding Samples for Okta CLI
```
Comment (mraible) [inline _source/_posts/2020-10-04-introducing-okta-cli.md:299]: Unfortunately, it's probably best to remove this section. Unless you want to offer an OktaDev t-shirt for creating a PR.

File: _source/_posts/2020-10-04-introducing-okta-cli.md [hunk 14]
```diff
@@ -0,0 +1,319 @@
...
+
+It's [Hacktoberfest](https://hacktoberfest.digitalocean.com/) and we'd love your contributions! If you want to setup a new sample, create a new repo that has the naming convention: `okta-<language and or framework name>-sample`. Ideally, it'd be a language and framework that we don't already have. But, if you want to submit a variation on an existing sample, we'd accept that too - just give it a unique name.
+
+Since Hacktoberfest runs off merged pull requests, we'll create the sample in the [okta-samples](https://github.com/okta-samples) GitHub org. You can then put up a pull request to it.
+
+## Wrapping Up
```
Comment (mraible) [inline _source/_posts/2020-10-04-introducing-okta-cli.md:305]:
```suggestion
## Learn More About Okta, OAuth 2.0, and OpenID Connect
```

**Review by mraible — CHANGES_REQUESTED** [review]
We might want to add a section with videos since we'll have three when this is published (Spring Boot, Angular, and React Native). Maybe a row with all three thumbnails? I can help with this.

File: _source/_posts/2020-10-04-introducing-okta-cli.md [hunk 15]
```diff
@@ -0,0 +1,319 @@
...
+    ".net",
+    "go",
+    "java",
+    "javascript",
+    "mobile",
+    "php",
+    "python",
+    "ruby",
+    "security",
+    "devops"
```
[the remark this answers is not in the document]
Comment (dogeared) [reply _source/_posts/2020-10-04-introducing-okta-cli.md:16]: It's aspirational - I want to create samples for all of these. Would you suggest I remove some?

File: _source/_posts/2020-10-04-introducing-okta-cli.md [hunk 16]
```diff
@@ -0,0 +1,319 @@
...
+]
+description: "Use the Okt CLI to get started in under 5 minutes!"
+tags: [security, oauth, oauth2, oidc, openid-connect]
+tweets:
+- "Add an Okta integration for your custom app in less than 5 minutes with the Okta CLI"
```
[the remark this answers is not in the document]
Comment (dogeared) [reply _source/_posts/2020-10-04-introducing-okta-cli.md:21]: not at this time, but that would be a great flag. Kind of like anyone can host start.spring.io. <- @bdemers

**Review by dogeared — COMMENTED** [review]
I addressed all the feedback, rebased master and renamed the file for tomorrow. The tables look reasonable on mobile.

image

image

**Comment by dogeared** [comment]
I addressed all the feedback, rebased master and renamed the file for tomorrow. The tables look reasonable on mobile.

image

image

**Comment by mraible** [comment]
Nice! You might want to copy it into GDoc for editorial or ask @KMDeMat to review this PR and add comments inline.

**Review by mraible — APPROVED** [review]

**Comment by mraible** [comment]
@dogeared I added screencasts in [this commit](https://github.com/oktadeveloper/okta-blog/pull/434/commits/22a82b4965d157fbaf517e557da3c1fae4fb0269). @Quorralyne I also added a link to your .NET 5 post. Please let us know if you'd rather have it link to a video (or update this branch with changes).

File: _source/_posts/2020-12-10-introducing-okta-cli.md [hunk 17]
```diff
@@ -51,7 +51,7 @@ brew cask install oktadeveloper/tap/okta
 
 Pretty easy, right?
 
-## Get Started in Under Five Minutes
+## Get Started in Under 5 Minutes
```
Comment (mraible) [inline _source/_posts/2020-12-10-introducing-okta-cli.md:54]: I think it looks better as a number in the title. :)

**Pull request merged**